app/aws/Federation.scala

```diff
@@ -1,31 +1,39 @@
 package aws
 
-import awscala.Credentials
 import awscala.iam._
-import awscala.sts.{FederationToken, STS, TemporaryCredentials}
-import com.amazonaws.auth.AWSStaticCredentialsProvider
 import com.amazonaws.services.identitymanagement.model.GetRoleRequest
-import com.amazonaws.services.securitytoken
 import com.gu.janus.model.{AwsAccount, Permission}
 import data.Policies
 import logic.Date
 import org.joda.time.{DateTime, DateTimeZone, Duration, Period}
+import play.api.libs.json.Json
+import software.amazon.awssdk.services.sts.StsClient
+import software.amazon.awssdk.services.sts.model.{
+  AssumeRoleRequest,
+  Credentials
+}
 
-import java.net.URLEncoder
+import java.net.{URI, URLEncoder}
+import java.nio.charset.StandardCharsets.UTF_8
+import scala.io.Source
 
 object Federation {
 
   /** Credential/Console lease times. Defaults are used when user doesn't
     * request specific time periods, max will limit how long can be requested.
     */
-  val maxShortTime = 3.hours
-  val minShortTime = 15.minutes
-  val defaultShortTime = 1.hour
-  val maxLongTime = 10.hours
-  val minLongTime = 1.hour
-  val defaultLongTime = 8.hours
+  val maxShortTime: Duration = 3.hours
+  val minShortTime: Duration = 15.minutes
+  val defaultShortTime: Duration = 1.hour
+  val maxLongTime: Duration = 10.hours
+  val minLongTime: Duration = 1.hour
+  val defaultLongTime: Duration = 8.hours
+
+  private val awsMinimumSessionLength = 900.seconds
 
-  val awsMinimumSessionLength = 900.seconds
+  private val signInUrl = "https://signin.aws.amazon.com/federation"
+  private val consoleUrl =
```

Comment:
Would it be clearer to apply the URLEncoding where it is used, to call this something like `encodedConsoleUrl`, or to inline this val?

This is only used in one place, but I can see the benefit to having it be defined up top as a "constant". However, including the encoding feels a little like mixing the constant definition with the logic of its use. I don't feel strongly, whatever you think!

Reply:
You're right that would be more consistent and readable.  It's a slight optimisation that's of very little value here.

Reply:
Done in b598b03b8a0b3905925cfbe5c1dea2313ce742cd